packages/affiliates/examples/dapp-mining/ethers-tagging.js

```diff
@@ -0,0 +1,57 @@
+const { ethers } = require("ethers");
+const { getAbi } = require("@uma/core");
+const argv = require("minimist")(process.argv.slice(), { string: ["emp", "tag", "tokens", "collateral"] });
+
+const {
+  // example emp for uusd weth
+  emp = "0x3605Ec11BA7bD208501cbb24cd890bC58D2dbA56",
+  // this should technically be a legit payout address for dapp mining rewards
+  tag = "0xa0dfF1F09a0DCEda7641b7eB07c3BDbe66a06C35",
+  // how many tokens to mint
+  tokens = "100",
+  // how much collateral to back the tokens
+  collateral = "100000"
+} = argv;
+
+// This is an example of creating a position in an EMP, the only function which needs to be tagged with your address
+function createData(empAddress, collateralToSend, tokensToCreate) {
+  const emp = new ethers.utils.Interface(getAbi("ExpiringMultiParty"));
+  return emp.encodeFunctionData("create", [
+    { rawValue: ethers.utils.parseUnits(collateralToSend) },
+    { rawValue: ethers.utils.parseUnits(tokensToCreate) }
+  ]);
+}
+
+// This takes encoded data field and just appends the tag
+// The tag must be valid hex
+function tagData(data, tag) {
+  return ethers.utils.hexConcat([data, tag]);
+}
+
+// Example of how you would construct the transaction with data field to be signed by private key
+function makeTransaction(data) {
+  return {
+    from: tag,
+    to: emp,
+    value: 0,
+    data,
+    gas: 1000000
+  };
+}
+
+function runExample() {
+  console.log("Running example with the following parameters:");
+  console.table({
+    emp,
+    tag,
+    tokens,
+    collateral
+  });
+  const data = createData(emp, collateral, tokens);
+  const taggedData = tagData(data, tag);
+  const transaction = makeTransaction(taggedData);
```

Comment:
do you want to send the transaction?

Reply:
I didnt want to actually send transaction in these examples otherwise to run it user would need to specify provider url, working emp address, etc. Just want to illustrate how to get 90% of the way there. But it might be useful to show how you would sign and send this, without actually doing it.

I think if i can abstract these into libraries id add some tests where the actual sending could be shown safely.

Reply:
I think its worth showing how to send it FWIW, we should make this as seamless as possible for the user to plug and play

Reply:
ye, I agree. this is not really a standard tx anymore so showing how to send it on-chain is a good idea.

Reply:
I've added some inline comments for showing how to send the transactions in both ethers and web3. I think this is probably the bare minimum to get the point across without having the script more complicated to run. Maybe i feel like its a lot to ask the user to run a local test node in order to run the script, but if you guys think that is an ok to assume i can make the script submit the TX as well

Reply:
sounds good to me!